avm-transpiler/scripts/compile_then_transpile.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+# This is a wrapper script for nargo.
+# Pass any args that you'd normally pass to nargo.
+# If the first arg is "compile",
+# run nargo and then transpile any created artifacts.
+#
+# Usage: compile_then_transpile.sh [nargo args]
+set -eu
+
+NARGO=${NARGO:-nargo}
+TRANSPILER=${TRANSPILER:-avm-transpiler}
+
+if [ "${1:-}" != "compile" ]; then
+  # if not compiling, just pass through to nargo verbatim
+  $NARGO $@
+  exit 0
+fi
+shift # remove the compile arg so we can inject --show-artifact-paths
+
+# Create a temporary file to capture and parse nargo's stdout while still printing it to the console.
+# To avoid a situation where the script fails and the temporary file isn't removed,
+# create file descriptors for writing/reading the temporary file and then remove the file.
+tmpfile=$(mktemp)
+exec 3>"$tmpfile"
+exec 4<"$tmpfile"
+rm "$tmpfile"
```

Comment:
Seems elaborate. Can we not just do something like? e.g.
`artifacts_to_transpile=$($NARGO compile --show-artifact-paths $@ | tee /dev/tty | grep -oP 'Saved contract artifact to: \K.*')`

Reply:
That might be fine, but then the user can't redirect nargo's stdout, right?